src/config/user_config.cpp

```diff
@@ -321,7 +321,154 @@ core::stringc ListUserConfigParam<T, U>::toString() const
     return "";
 }   // toString
 
+    // ============================================================================
+template<typename T, typename U>
+MapUserConfigParam<T, U>::MapUserConfigParam(const char* param_name,
+    const char* comment)
+{
+    m_param_name = param_name;
+    all_params.push_back(this);
+    if (comment != NULL) m_comment = comment;
+}   // MapUserConfigParam
 
+    // ============================================================================
+template<typename T, typename U>
+MapUserConfigParam<T, U>::MapUserConfigParam(const char* param_name,
+    const char* comment,
+    int nb_elements,
+    ...)
+{
+    m_param_name = param_name;
+    all_params.push_back(this);
+    if (comment != NULL) m_comment = comment;
+
+    // add the default list
+    va_list arguments;
+    va_start(arguments, nb_elements);
+   
+    struct pair_type { T key; U value; };
+
+    for (int i = 0; i < nb_elements; i++) {
```

Comment:
Braces go on a newline, here and in many other places

Reply:
I'll fix it in other places, however when defining the struct which is equivalent to an std::pair<T,U> is it acceptable to keep it all in one line considering that's how an std::pair would be? (Also it would take up significantly less space).

Reply:
You mean something like "struct X { int a; int b; }"? yes that would be OK